Theory assessment/Theory_assessment_Software_Engineering.md

```diff
@@ -0,0 +1,150 @@
+### 1. How does Object-oriented Programming differ from Process Oriented Programming? 
+A: Object-oriented programming is divided into the objects, but process-oriented programming is divided into smaller functions. OOP has bottom-up approach and data members can be public/private/default, which can also help hide the data and is more secure. POP has top-to-down approach and the data members are public, also data hiding is impossible. 
+Function overloading in OOP is possible, but not possible in POP. OOP is great for solving big problems, and it focuses on data, POP focuses on functions.
+
+### 2. What's polymorphism in OOP?
+A: Polymorphism in OOP means you can use same function for different purposes, used and reused many times. Example: you can overwrite something that has different outcome in parent class. Has positive influence to code, because can save time and makes easy to debug if needed.
+```
+class Animals:
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        
+    def lives_in_farm(self):
+        return True
+            
+class Dog(Animals):
+    def __init__(self, name, age):
+        super().__init__(name, age)
+        
+    def lives_in_farm(self):
+        return False
+
+dog_Mika=Dog('Mika', 3)
+print(dog_mika.name, dog_mika.age, dog_mika.lives_in_farm())
+```
+### 3. What's inheritance in OOP?
+A: Inheritance in OOP means that we can transfer methods/properties from one class to another. It helps with code reading, reusing and scaling and decreases repetition. Also better and easier to find errors.
```

Comment:
7/10 there are types of inheritance, which ideally should be described here. i.e. single, hierarchy, multi-level, etc.